src/zebras/_period.py

```diff
@@ -339,10 +343,29 @@ def _encode_zebras_periods(v: Periods):
     }
 
 
-def period(data, freq=None) -> Period:
-    if hasattr(data, "freqstr") and freq is None:
-        freq = Freq.from_pandas(data.freqstr)
-    else:
+def period(
+    data: Union[Period, str], freq: Optional[Union[Freq, str]] = None
+) -> Period:
+    """Create a `zebras.Period` object that represents a period of time.
+
+    Parameters
+    ----------
+    data
+        The time period represented by a string (e.g., "2023-01-01"),
+        or another Period object.
+    freq, optional
+        The frequency of the period, e.g, "H" for hourly, by default None.
+
+    Returns
+    -------
+        A `zebras.Period` object.
+    """
+    if freq is None:
+        if hasattr(data, "freqstr"):
+            freq = Freq.from_pandas(data.freqstr)
+        else:
+            raise ValueError("No frequency specified.")
+    elif isinstance(freq, str):
         freq = Freq.from_pandas(freq)
```

Comment:
I think we might want to add an explicit test for `Freq` here.

Reply:
Updated but not sure about the `ValueError`. Is this okay?

Reply:
I think `ValueError` is fine.